pkg/kubelet/dockertools/docker.go

```diff
@@ -1049,3 +1049,21 @@ func RunContainer(client DockerInterface, container *api.Container, pod *api.Pod
 	}
 	return dockerContainer.ID, nil
 }
+
+// KillContainer kills a docker container with the given containerID.
+// TODO(yifan): To use strong type for the containerID.
+func KillContainer(client DockerInterface, containerID string,
```

Comment:
@vmarmol I have spread thoughts here: https://github.com/GoogleCloudPlatform/kubernetes/pull/6100#issuecomment-87109691

Think we can just go that way, still let container runtime to handle refs, readiness for now, but move them out of the RunContainer() or KillContainer(). We can come back to revisit this later. How do you think about that?